.travis.yml

```diff
@@ -32,38 +32,70 @@ jobs:
       #if: tag =~ /^release.*$/
       language: shell
       services: docker
+      install:
+        # set up awscli packages
+        - python3 -m pip install --user awscli
+      before_script:
+        # set up shared storage for build artifacts
+        - mkdir -p $TRAVIS_BUILD_SHARED_STORAGE
       script:
         - docker run --rm -v `pwd`:/var/task amazon/aws-sam-cli-build-image-python3.8 bash -c './build-release-package.sh'
-        - cp -p dynatrace-aws-log-forwarder.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+      after_success:
+        - cp -p dynatrace-aws-log-forwarder.zip $TRAVIS_BUILD_SHARED_STORAGE
+        - aws s3 sync $TRAVIS_BUILD_SHARED_STORAGE $AWS_S3_BUILD_SHARED_STORAGE
     - stage: e2e_tests
       name: Execute e2e tests
       if: branch = master OR type = pull_request
       language: shell
       services: docker
-      before_script: 
-        - mkdir -p ./e2e_tests_tmp
-        - cp -p ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER/dynatrace-aws-log-forwarder.zip ./e2e_tests_tmp
-        - cd ./e2e_tests_tmp
+      install:
+        # set up awscli packages
+        - python3 -m pip install --user awscli
+        # set up e2e packages
+        - python3 -m pip install --user -r ./tests/e2e/requirements.txt
+      before_script:
+        # set up shared storage for build artifacts
+        - mkdir -p $TRAVIS_BUILD_SHARED_STORAGE
+        - aws s3 sync $AWS_S3_BUILD_SHARED_STORAGE $TRAVIS_BUILD_SHARED_STORAGE
+        - cp -p $TRAVIS_BUILD_SHARED_STORAGE/dynatrace-aws-log-forwarder.zip ./tests/e2e
+        - cd ./tests/e2e
         - unzip dynatrace-aws-log-forwarder.zip
       script:
         # deploy log forwarder resources reqiured for e2e to AWS
-        - ./dynatrace-aws-logs.sh deploy --target-url $DYNATRACE_ENV_URL --target-api-token $DYNATRACE_API_KEY --use-existing-active-gate true --require-valid-certificate $VERIFY_SSL --stack-name aws-log-forwarder-e2e-build-$TRAVIS_BUILD_NUMBER
+        - ./dynatrace-aws-logs.sh deploy --target-url $DYNATRACE_ENV_URL --target-api-token $DYNATRACE_API_KEY --use-existing-active-gate true --require-valid-certificate $VERIFY_SSL --stack-name $AWS_CLOUDFORMATION_STACK
+        # create AWS CloudWatch Logs log groups and assign log streams

```

Comment:
Those descriptive comments are not needed, very easy to read this from the code

Reply:
For me, let them be, we suffer from too few comments in that repo as it is.

Reply:
Comments of type "create log groups" where next line says "aws logs create-log-group" bring no value at all from my perspective. I am all for comments but their role is to bring context and explain things, not repeat what next line says

Reply:
Brevity of comment vs the actual mouthful to read in the next lines makes a difference to me. Just making a second point here, don't intend to discuss it much further.

Reply:
If at least two people find them useful (Beata and Kamil), I'd leave them. It's not like they obscure the code or anything. They may be not as useful as
```
# include $HOME/.local/bin for `aws`
```
but I think they do no harm